### PR TITLE
Sagemaker Hyperpod Training Adaptor For Nemo Release 1.1.1

### PR DESCRIPTION
**Pull request by julianhr**
### Motivation
Fix problems running recipes on Kubernetes in a p5 HyperPod environment

### Changes
* Remove support for the SMDDP library. Only NCCL is now accepted as a backend.

### Testing
Extensively tested during development

## Merge Checklist
Put an x in the boxes that apply. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request.

### General
 - [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
 - [x] I have run `pre-commit run --all-files` on my code. It will check for [this configuration](../.pre-commit-config.yaml).
 - [x] I have updated any necessary documentation, including [READMEs](../README.md) and API docs (if appropriate)
 - [x] I have verified the licenses used in the license-files artifact generated in the Python License Scan CI check. If the license workflow fails, kindly check the licenses used in the artifact.

### Tests
 - [x] I have run `pytest` on my code and all unit tests passed.
 - [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
 - [x] I have ran a training job with real data (huggingface dataloader instead of synthetic data) and validated that the loss value goes down.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/hyperpod_nemo_adapter/collections/parts/fsdp_strategy.py [hunk 1]
```diff
@@ -97,13 +91,8 @@ def __init__(
         self.smp_config_dict = self._setup_smp_config(cfg)
         self.app_state = SageMakerAppState()
 
-        # Init from original PT-Lightning policy to avoid megatron specific initialization
+        # Init from original PT-Lightening policy to avoid Megatron specific initialization
```
Comment (crazyguitar) [inline src/hyperpod_nemo_adapter/collections/parts/fsdp_strategy.py:94]: I think lightning is correct one.

https://github.com/Lightning-AI/pytorch-lightning

Comment (julianhr) [reply src/hyperpod_nemo_adapter/collections/parts/fsdp_strategy.py:94]: updated

**Review by jessech-en — APPROVED** [review]

**Review by haohanchen-aws — APPROVED** [review]
